### PR TITLE
fix(citus-cdc): fix publication alter of citus connector

### PR DESCRIPTION
**Pull request by StrikeW**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
citus connector will create a FOR ALL TABLES publication, which will replicates changes for all tables in the database, including tables created in the future. And it cannot be altered afterwards.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: integration_tests/citus-cdc/create_source.sql [hunk 1]
```diff
@@ -20,5 +20,30 @@ CREATE TABLE orders_rw (
     schema.name = 'public',
     table.name = 'orders',
     slot.name = 'orders_dbz_slot',
-    publication.create.enable = 'true'
```
Comment (tabVersion) [inline integration_tests/citus-cdc/create_source.sql:23]: so `publication.create.enable = 'true'` is a default behavior?

**Review by tabVersion — APPROVED** [review]
thanks for your effort

**Comment by tabVersion** [comment]
and can you explain why we need this change? I am not familiar with citus context

**Comment by StrikeW** [comment]
> and can you explain why we need this change? I am not familiar with citus context

Our citus connector will create a `FOR ALL TABLES` publication to replicate all sharding tables (table_001,t able_002, etc.) for each worker node. This publication cannot be altered afterwards, otherwise error will occurs. This PR fix the issue.

File: integration_tests/citus-cdc/create_source.sql (hunk 1 above)
Comment (StrikeW) [reply integration_tests/citus-cdc/create_source.sql:23]: Yes, user can create and provide their own pub.

**Pull request merged**